src/app/models/AbstractModel.ts

```diff
@@ -82,7 +82,15 @@ export abstract class AbstractModelWithoutId<Model = Record<string, any>> {
    * be sent in a JSON API request
    */
   public getJsonAttributes(opts?: ModelSerializationOptions): Partial<this> {
-    return this.toObject(this.getModelAttributes(opts), opts);
+    const keys = this.getModelAttributes(opts);
+    const data = this.toObject(keys, opts);
+
+    // if there is an image object attached to the request, remove it
+    if (!isInstantiated(data["image"])) {
+      delete data["image"];
+    }
```

Comment:
```suggestion
    // if there is an image object attached to the request, remove it
    if (!isInstantiated(data["image"])) {
      delete data["image"];
    }
```

this is the wrong place for this change.

also special casing on stringly typed attribute is bad style 

also your comment explains what the code is doing, and not why.